benchmarks/integer_overflow_and_underflow/overflow_add.yaml

```diff
@@ -0,0 +1,8 @@
+integer_underflow_and_overflow/overflow_add:
+  has_bug: true  # Trail of bits suggests adding a test where has_bug: false
```

Comment:
Not sure why we need this. We could have a benchmark that zero issues

Reply:
Yes. Look at Suhabe suite and there are deceptively-close cases. So much that we called them "benign". See also https://discourse.secureth.org/t/building-an-ethereum-security-benchmark/63 and quoting from that: 

> Each test should include an annotated correct or non-vulnerable variant

Reply:
Yes, that is what @vl-ti would has proposed and would like to do.

Reply:
I think this is overcomplicating things. If there's a "benign" case, it should simply not be in the list of expected issues.

E.g. there's an input file with a non-reachable integer overflow, the expected result is an empty list. If a tool reports an integer overflow or anything else, it's a false positive.

Reply:
Wait a minute - now we are changing topics. has_bug is about whether there is a bug or not. Not all benchmarks should have bugs in them. 

Some as in the Suhabe benchmark suite have these global properties like the sum of values in a dynamic array is less than the amount I put in at the begining. Mytrhil (among others) doesn't detect this global property so it sees and an Integer Overflow. "has_bug" is explicit that there is no bug in this particular kind of benchmark.

The benign thing is a different matter.

Reply:
If there's no bug then the list of expected issues in the benchconf is empty. No issues expected implies that the contract has no bug so an explicit `has_bug` variable is not needed.

Reply:
A "benign" issue is no bug and should no be part of the expected outputs. If the tool reports the benign issue it fails the test.

Reply:
I would prefer to be explicit. It is too easy to leave the issues link empty. In fact often that is how things start out in benchmarks were we don't know what is up.

Issues have to be somewhat customized to the tool that is used - bytecode if the tool only deals with that and source location if the tool only uses that.

Reply:
> In fact often that is how things start out in benchmarks were we don't know what is up.

That's not a good thing. We should know exactly what's up for each benchmark we add. Currently, it appears like we're running tools and putting whatever the output is into the tool-specific config files. For example, the [Oyente benchconf](https://github.com/EthereumAnalysisBenchmarks/ethereum-analyzer-suites-runner/blob/master/benchconf/nssc-Oyente.yaml) for Gift box contains lots of non-existent integer overflows.

If we aren't sure about what the vulnerabilities in a particular file are then we can't include it in the benchmark. 

> It is too easy to leave the issues link empty. 

Isn't it just as easy to set `has_bugs` to false?

Reply:
> Isn't it just as easy to set has_bug to false?

No it isn't. Looking at the benchmark it is often easy to determine if there should be a bug or not. 

_Is a lot less easy to determine **what**  the problem or problems are going to be for a particular tool._

Reply:
> Is a lot less easy to determine what the problem or problems are going to be for a particular tool.

Every benchmark is associated with an objective list of security issues is contains. Either the tool reports them correctly or it doesn't.

Wasn't the consensus between Geri and you that we'd come up with an universal standard? Anybody who wants to use our benchmark can add standard-conform output to their tool. If they don't want to be benchmarked that's also fine. I'm not sure it's our task to do implement support for every tool out there and "force" them to be measured against the benchmark.

Reply:
Let's plan on discussing this when we meet in person.